addons/base_import/models/base_import.py

```diff
@@ -577,7 +579,14 @@ def _read_csv(self, options):
             if bom and csv_data.startswith(bom):
                 encoding = options['encoding'] = encoding[:-2]
 
-        csv_text = csv_data.decode(encoding)
+        try:
+            csv_text = csv_data.decode(encoding)
+        except UnicodeDecodeError as exc:
+            raise ImportValidationError(_(
+                "There was an issue decoding the file using encoding '%(encoding)s'. %(extra)s",
+                encoding=encoding,
+                extra=_("This encoding was automatically detected.") if encoding_guessed else _("This encoding was manually selected."),
+            )) from exc
```

Comment:
Please refrain from composing translations yourself. The %(extra)s part makes sense in English but there is no guarantee the sentence will be structured the same way in all languages (think arabic and cjk). For people doing translations, they will thing the string "There was an issue..." and "This encoding was..." at very remote places in their translation software, making it difficult for them to effectively translate the terms (they'll need to have a look at the source code, but they are not developers).

Also we have a guideline to use curly-quotes “...” instead of single/double quotes. See odoo/tools/translates.py.

```suggestion
            if encoding_guessed:
                msg = _("There was an issue decoding the file using encoding “%(encoding)s”.\nThis encoding was automatically detected.", encoding)
            else:
                msg = _("There was an issue decoding the file using encoding “%(encoding)s”.\nThis encoding was manually selected.", encoding)
            raise ImportValidationError(msg) from exc
```